web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx

```diff
@@ -320,10 +331,43 @@ function sortResourcesByKind(
   return sorted;
 }
 
-// Sort the resources alphabetically and with the Guided resources listed first.
-export function sortResources(resources: ResourceSpec[]) {
+// Sort the resources by 1. preferred 2. guided and 3. alphabetically
+export function sortResources(
+  resources: ResourceSpec[],
+  preferences: UserPreferences
+) {
+  // A user can have preferredResources set via their onboarding survey.
+  // We sort the list by the preferred resource type but do not apply a search.
+  const preferredResources =
+    (preferences &&
+      preferences.onboard &&
+      preferences.onboard.preferredResources) ||
+    [];
+  const hasPreferredResources =
+    preferredResources && preferredResources.length > 0;
+  const maxResources = Object.keys(ClusterResource).length / 2 - 1;
```

Comment:
```suggestion
  const maxNumPreferredResources = Object.keys(ClusterResource).length / 2 - 1;
```

Reply:
> the ordering won't probably be too obvious (or makes sense) to the user

This is very fair, but we don't have any new designs in scope this quarter. I know Kenny is thinking of ways to make this clearer though.